tools/ipo/macros.xml

```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0"?>
+<macros>
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="1.10.0">bioconductor-ipo</requirement>
+            <requirement type="package" version="1.1_5">r-batch</requirement>
+        </requirements>
+    </xml>
+    <xml name="stdio">
+        <stdio>
+            <exit_code range="1" level="fatal" />
+        </stdio>
+    </xml>
+
+    <token name="@COMMAND_SCRIPT@">
+        LANG=C Rscript $__tool_directory__/
+    </token>
+
+    <token name="@COMMAND_LOG_EXIT@">
+        ;
+        return=\$?;
+        mv log.txt $log;
+        cat $log;
+        sh -c "exit \$return"
+    </token>
+
+    <token name="@COMMAND_NSLAVES@">
+        nSlaves \${GALAXY_SLOTS:-1}
+    </token>
+
+    <!-- zipfile load for planemo test -->
+
+    <token name="@COMMAND_FILE_LOAD@">
+        #if $file_load_section.file_load_conditional.file_load_select == "yes":
+            #if $file_load_section.file_load_conditional.input[0].is_of_type("mzxml") or $file_load_section.file_load_conditional.input[0].is_of_type("mzml") or $file_load_section.file_load_conditional.input[0].is_of_type("mzdata") or $file_load_section.file_load_conditional.input[0].is_of_type("netcdf"):
+                #set singlefile_galaxyPath = ','.join( [ str( $single_file ) for $single_file in $file_load_section.file_load_conditional.input ] )
+                #set singlefile_sampleName = ','.join( [ str( $single_file.name ) for $single_file in $file_load_section.file_load_conditional.input ] )
+
+                singlefile_galaxyPath '$singlefile_galaxyPath' singlefile_sampleName '$singlefile_sampleName'
```

Comment:
singlefile_sampleName --> needs to be quoted I assume ... it can contain whitespaces.

$single_file.name a user could create a file with a name `'rm -rf /'` or something like that https://github.com/galaxyproject/tools-iuc/blob/9b67be72f7330b1b9af7944aa695c9f5d6d20a15/tools/intervene/macros.xml#L34  ...